lib/OpenQA/Worker/Job.pm

```diff
@@ -445,7 +445,8 @@ sub _stop_step_4_upload ($self, $reason, $callback) {
             my @other = (
                 @{find_video_files($pooldir)->map('basename')->to_array},
                 COMMON_RESULT_FILES,
-                qw(serial0 video_time.vtt serial_terminal.txt virtio_console.log virtio_console1.log)
+                # TODO: use $vars->{VIRTIO_CONSOLE_NUM} for virtio names
+                qw(serial0 video_time.vtt serial_terminal.txt virtio_console.log virtio_console1.log virtio_console_user.log)
```

Comment:
Since the additional file is generated by the os-autoinst feature you're developing in https://github.com/os-autoinst/os-autoinst/pull/2240 it would be good to add a comment in the code about it (to make it clear under which circumstances this special "user" file is actually generated).

Reply:
Sure, I can add a comment, although nobody cared in the past (all these files are created from os-autoinst.
What exactly do you want me to add?

To be honest, the best would be some constant e.g. svirt uses:

https://github.com/os-autoinst/os-autoinst/blob/b66759aea2b005800a00ddf5f8a405fc3c8b496b/backend/svirt.pm#L12-L13

https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/9f84f058d22425bae123734c06a28fa988d5ba06/lib/bootloader_setup.pm#L1126-L1131

Because constant is much easily searchable. But I don't know where should be this constant to be reusable with both code in https://github.com/os-autoinst/os-autoinst/ and https://github.com/os-autoinst/openQA/ (here).

I actually planned it, but then decide to postpone it (I've already touched quite few things).

Reply:
True. However, there should be *some* documentation what the `_user`-suffixed files mean. Of course it isn't optimal that we maybe don't even have documentation for other files as well.

Not sure about the constant considering that sharing it between the code bases isn't that easy, indeed.

Reply:
Well, undocumented strings are in more places, e.g.:

https://github.com/os-autoinst/openQA/blob/582a0875e3b64c4c959be537b2cc7f74c00b5308/lib/OpenQA/Worker/Job.pm#L447-L448

https://github.com/os-autoinst/openQA/blob/582a0875e3b64c4c959be537b2cc7f74c00b5308/lib/OpenQA/Schema/Result/Jobs.pm#L1130

https://github.com/os-autoinst/openQA/blob/582a0875e3b64c4c959be537b2cc7f74c00b5308/lib/OpenQA/Schema/Result/Jobs.pm#L1144

IMHO best would be do document the file at git commit log (searchable in git blame) and do a proper cleanup (as a separate effort) - put more constants in

https://github.com/os-autoinst/openQA/blob/582a0875e3b64c4c959be537b2cc7f74c00b5308/lib/OpenQA/Jobs/Constants.pm#L91

where their meaning would be described. If you insist on documenting it in the code here, I'll document everything but `video_time.vtt` which I don't have idea what  (looks like something for video), so that I omit that (unless you specify what to add).

Reply:
I don't insist :-)

Reply:
> Not sure about the constant considering that sharing it between the code bases isn't that easy, indeed.

Constants could be either shared via https://github.com/os-autoinst/os-autoinst-common (maybe danger) or defined in both repos (os-autoinst and openQA) - that's not optimal, but searching constant is much better than search plain text. 

Maybe I'm too influenced by C preprocesor's `#define`, but IMHO constants for string paths, names, etc. should have been here from the start. You find a definition in obvious file `Constants.pm` and read a brief documentation about purpose of the value (thus this doc does not have to be repeated all over).

Reply:
@Martchus in the end I at least added one commit, where I documented at least `serial_terminal.txt` being added vs. `virtio_console.log` being renamed to `serial_terminal.txt`. Even I (the original author) forgot it :).